src/Behat/Behat/Output/Node/EventListener/JUnit/JUnitOutlineStoreListener.php

```diff
@@ -53,7 +54,18 @@ public function __construct(SuitePrinter $suitePrinter)
      */
     public function listenEvent(Formatter $formatter, Event $event, $eventName)
     {
-        $this->captureOutlineOnBeforeOutlineEvent($event);
+      $this->captureOutlineOnBeforeOutlineEvent($event);
+      if ($event instanceof SuiteTested) {
+        foreach ($event->getSpecificationIterator() as $node) {
```

Comment:
Can you isolate the `foreach` loop into a dedicated method on the class? An explicit name (as you can see on other methods of this class) is required tho